provider/runbook_data.go

```diff
@@ -53,10 +65,16 @@ func dataFireHydrantRunbook(ctx context.Context, d *schema.ResourceData, m inter
 		return diag.Errorf("Error reading runbook %s: %v", runbookID, err)
 	}
 
+	attachmentRule, err := json.Marshal(runbookResponse.AttachmentRule)
+	if err != nil {
+		return diag.Errorf("Error converting step config to JSON due invalid JSON returned by FireHydrant: %v", err)
+	}
```

Comment:
could this move down a lil to below line 79 to keep all the reading and setting logic together?

to make the read here match the suggestion below and account for the fact that we have to normalize the JSON string ourselves, this will need to change to something more like this:
```suggestion
	var attachmentRule []byte
	if len(runbookResponse.AttachmentRule) > 0 {
		attachmentRule, err = json.Marshal(runbookResponse.AttachmentRule)
		if err != nil {
			return diag.Errorf("Error converting attachment rule to JSON due invalid JSON returned by FireHydrant: %v", err)
		}
	}
	normalizedAttachmentRuleJSON, _ := structure.NormalizeJsonString(string(attachmentRule))
	attributes["attachment_rule"] = normalizedAttachmentRuleJSON
```